Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm

```diff
@@ -345,6 +345,10 @@
         [webView _uninstallScreenTimeWebpageController];
         return;
     }
+
+    if (pageProxy)
+        pageProxy->makeViewBlankIfUnpaintedSinceLastLoadCommit();
```

Comment:
Isn't this going to make the view blank every time you navigate to a new page, regardless of whether the blocking view was shown?

Reply:
Yeah, I was just about to talk to you about it lol. I think maybe smth like if URLIsBlocked --> make view blank before navigating

Reply:
I also was gonna ask about the pageProxy check (since I check it before as well so it's redundant)

Reply:
> I also was gonna ask about the pageProxy check

You can probably return early (uninstalling the ST controller if needed), if the page is null.